### PR TITLE
User story3

### PR DESCRIPTION
**Pull request by deriancastro**
Hi, here is my third user-story, now die App has a goals page and can be navigated, I will be glad for your reviews - be kind :)

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -18,30 +18,61 @@ export default function App() {
     'Kata guruma',
   ]
 
+  const goalsList = [
```
Comment (badabam) [inline src/App.js:21]: Please think about making this a JSON-File. You could directly add an `id` for each entry like that:
```js
  { text: 'Improve the performance of my left uchimata', id: '1' },
  { text: 'See the yoko tomoe tutorial', id: '2' },
    ```

File: src/App.js [hunk 2]
```diff
@@ -18,30 +18,61 @@ export default function App() {
...
+
   return (
     <div>
       {currentPage === 'tutorialPage' && (
         <TutorialPage
           pageName="TUTORIAL"
           techNamesList={techNamesList}
-          toDetailPage={handleToDetailPage}
+          onDetail={showDetailPage}
+          onNavigate={showGoalsPage}
```
Comment (badabam) [inline src/App.js:46]: Good wording!

File: src/App.js [hunk 3]
```diff
@@ -18,30 +18,61 @@ export default function App() {
...
   }
-  function handleToTutorialPage() {
+  function showTutorialPage() {
     setCurrentPage('tutorialPage')
   }
+  function showGoalsPage() {
+    setCurrentPage('goalsPage')
+  }
+  function handleIsChecked() {
+    return console.log('The data will save in Localstorage')
```
Comment (badabam) [inline src/App.js:76]: This looks unfinished. Save it in localStorage and don't console.log about it. 😉

File: src/components/Button.spec.js [hunk 4]
```diff
@@ -3,8 +3,10 @@ import userEvent from '@testing-library/user-event'
 import Button from './Button'
 
 describe('Button', () => {
+  const noop = () => {}
+
   it('contains a text', () => {
-    render(<Button onClick={jest.fn()}>Back</Button>)
+    render(<Button onClick={noop}>Back</Button>)
```
Comment (badabam) [inline src/components/Button.spec.js:9]: Yes, exactly!

File: src/components/Card.js [hunk 5]
```diff
@@ -3,12 +3,12 @@ import PropTypes from 'prop-types'
 
 Card.propTypes = {
   techName: PropTypes.string,
-  toDetail: PropTypes.func,
+  onDetail: PropTypes.func,
```
Comment (badabam) [inline src/components/Card.js:6]: Should have `.isRequired`

File: src/components/Goal.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
+import styled from 'styled-components/macro'
+import PropTypes from 'prop-types'
+
+Goal.propTypes = {
+  goalText: PropTypes.string,
+  goalNumber: PropTypes.number,
+  isChecked: PropTypes.func,
```
Comment (badabam) [inline src/components/Goal.js:7]: `isChecked` sounds like a boolean (it's a convention to prefix booleans with `is`, `has`, `does`). Please rename it. e.g. `onCheckGoal`

File: src/components/Goal.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+export default function Goal({ goalText, goalNumber, isChecked }) {
+  return (
+    <Wrapper data-testid="goal">
+      <ItemNumber data-testid="itemNumber">{goalNumber}.</ItemNumber>
+      <CheckBox>
+        <Text>{goalText}</Text>
+        <input onChange={isChecked} type="checkbox" aria-checked="true" />
```
Comment (badabam) [inline src/components/Goal.js:16]: `aria-checked` should get different values (from a boolean called `isChecked` maybe?).

File: src/pages/GoalsPage.js [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+  isChecked,
+  onNavigate,
+}) {
+  return (
+    <Wrapper>
+      <Header>{pageName}</Header>
+      <Container>
+        <List>
+          {goalsList.map(({ text }, index) => (
+            <li key={index}>
```
Comment (badabam) [inline src/pages/GoalsPage.js:26]: Better give the goals some ids and use them as the `key`.

File: src/pages/GoalsPage.js [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+  )
+}
+
+const Wrapper = styled.div`
+  display: grid;
+  grid-template-rows: 60px auto 50px;
+  gap: 3px;
+  height: 100vh;
+`
+const Container = styled.section`
```
Comment (badabam) [inline src/pages/GoalsPage.js:49]: Maybe call it `Scroller` or `ScrollContainer` because that's the special thing about it?

**Review by badabam — CHANGES_REQUESTED** [review]
You improved quite a few things, well done! But one thing _needs_ to be changed: the prop `isChecked` should be named different (see comments) as it is no boolean. And you **also** need a boolean `isChecked` that you can use for the `aria-checked`. 

Please fix that and push it in this PullRequest.

File: src/pages/DetailPage.js [hunk 10]
```diff
@@ -2,22 +2,22 @@ import styled from 'styled-components/macro'
...
 }
 
-export default function DetailPage({ toTutorialPage, pageName }) {
+export default function DetailPage({ onNavigate, pageName }) {
   return (
     <Wrapper>
       <Header>{pageName}</Header>
       <Container>
-        <img src={judo}></img>
+        <img src={judoImage} alt=""></img>
```
Comment (Tim-Pet) [inline src/pages/DetailPage.js:17]: An alt text would be nice

File: src/App.js (hunk 3 above)
Comment (Tim-Pet) [inline src/App.js:76]: If you just console log in a function the user probably won't have a benefit from that function => implement the function only if you need it for the current us.

File: src/pages/TutorialPage.spec.js [hunk 11]
```diff
@@ -17,41 +19,48 @@ describe('TutorialPage', () => {
...
-    expect(itemList[0]).toHaveTextContent('Uchimata')
-    expect(itemList[1]).toHaveTextContent('Harai goshi')
-    expect(itemList[2]).toHaveTextContent('Seoi nage')
-    expect(itemList[3]).toHaveTextContent('Ko uchi gari')
-    expect(itemList[4]).toHaveTextContent('O soto gari')
-    expect(itemList[5]).toHaveTextContent('Kubi nage')
-    expect(itemList[6]).toHaveTextContent('Yoko tomoe')
-    expect(itemList[7]).toHaveTextContent('Kata guruma')
+    const cardList = screen.getAllByTestId('card')
+    expect(cardList).toHaveLength(8)
```
Comment (Tim-Pet) [inline src/pages/TutorialPage.spec.js:32]: nice! 🚀

**Review by Tim-Pet — COMMENTED** [review]
Nice one - I guess Jerries comments will do the trick here. Can't do much but agree with his comments 🥳

File: src/App.js (hunk 1 above)
Comment (deriancastro) [reply src/App.js:21]: Done and I add a "isChecked" too

File: src/App.js (hunk 2 above)
Comment (deriancastro) [reply src/App.js:46]: good:)

File: src/components/Goal.js (hunk 7 above)
Comment (deriancastro) [reply src/components/Goal.js:16]: Done, but I thought, that by render I could see some checkboxes 'checked':
Bildschirmfoto 2021-06-08 um 23 05 26

File: src/pages/GoalsPage.js (hunk 9 above)
Comment (deriancastro) [reply src/pages/GoalsPage.js:49]: done

Comment (deriancastro) [reply src/pages/GoalsPage.js]: the commit with the changes is at the end, after the dialogues

File: src/components/Goal.js (hunk 7 above)
Comment (deriancastro) [reply src/components/Goal.js]: It works just with 'checked'
Bildschirmfoto 2021-06-09 um 00 02 20

File: src/App.js [hunk 12]
```diff
@@ -4,10 +4,10 @@ import DetailPage from './pages/DetailPage'
 import GoalsPage from './pages/GoalsPage'
 
 export default function App() {
+  const goalsJSON = require('./data/goalsList.json')
   const [currentPage, setCurrentPage] = useState('tutorialPage')
   const [currentTechName, setCurrentTechName] = useState('')
-
-  const goalsList = require('./data/goalsList.json')
+  const [goalsList, setGoalsList] = useState(goalsJSON)
```
Comment (badabam) [inline src/App.js]: Please use `import` for importing your json file as well.

File: src/App.js [hunk 13]
```diff
@@ -56,7 +56,12 @@ export default function App() {
...
   }
-  function handleOnCheckGoal() {
-    return console.log('The data will save in Localstorage')
+  function handleGoal(index) {
+    const goalToUpdate = goalsList[index]
+    setGoalsList([
+      ...goalsList.slice(0, index),
+      { ...goalToUpdate, isChecked: !goalToUpdate.isChecked },
+      ...goalsList.slice(index + 1),
+    ])
```
Comment (badabam) [inline src/App.js:65]: Yes!

**Review by badabam — APPROVED** [review]
Looks nice! Please update the json file import and then 🚢 it!

File: src/App.js (hunk 13 above)
Comment (deriancastro) [reply src/App.js:65]: OK, but Morten had to help me understand the mechanism

**Pull request merged**